scripts/demo_common.sh

```diff
@@ -63,6 +80,9 @@ while true; do
   esac
 done
 
+# Now that we have a root path we can source the "project_detection.sh" script.
+. "$root_dir/framework/scripts/project_detection.sh"
```

Comment:
I'm not familiar with the history here. Have you considered having a `demo_common.sh` in each branch that sources a `demo_framework.sh` in the framework?

Reply:
Do you mean having a `demo_common.sh` in `tf-psa-crypto/programs/psa` for `key_ladder_demo` and `psa_hash_demo` and one in `frameworks/tests/programs` for `dlopen_demo`?
That's an option. The only drawback I see is the partial duplication of the initial `while` loop to search for the root folder. Not full duplication because there would be differences, but is it worth it?
If you prefer that way just let me know and I will implement this path ;)

Reply:
I don't really understand what is the proposition here and what problem we are trying to solve.

Reply:
We have:

1. Demo scripts in consuming branches.
2. Common code to all demo scripts.
3. Code that wants to know which branch it's consumed by.

Currently the common code is in `framework/scripts/demo_common.sh`, and it has somewhat complicated code to both detect the project root and identify the branch, with some special casing because some scripts want to find mbedtls-parenting-crypto when run from a crypto script.

My alternative proposal is to have an intermediate script in each branch that does the branch-specific setup, locates the framework and calls the common code in the framework. This way, no branch detection is necessary.

Reply:
Okay but with the proposal in https://github.com/ronald-cron-arm/mbedtls-framework/tree/pr159-test based on [one of your proposal](https://github.com/Mbed-TLS/mbedtls-framework/pull/159#discussion_r2040101717) (or at least my interpretation of it) the project root detection is not branch specific anymore. We have only one temporary trick for the path to `query_compile_time_config` that we can remove when support for `query_compile_time_config` is added in TF-PSA-Crypto. We have https://github.com/Mbed-TLS/TF-PSA-Crypto/issues/127 tracking that work.

Reply:
@ronald-cron-arm I just updated my commit including changes from your branch.